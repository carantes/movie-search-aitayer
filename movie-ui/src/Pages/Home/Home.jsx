import React from "react";
import { ClipLoader } from "react-spinners";
import PropTypes from "prop-types";
import { Layout, Search, MovieList } from "Components";
import { withMovies } from "HOC";
import { WAIT_INTERVAL, TYPE_COUNT } from "Utils";

const { Wrapper, Header, Main } = Layout;

export const Home = ({ movies, loading, getMovies }) => (
  <Wrapper>
    <Header title="Movie Search">
      <Search
        onSearch={keyword => getMovies(keyword)}
        startWith={TYPE_COUNT}
        waitInterval={WAIT_INTERVAL}
      />
    </Header>
    <Main>
      <ClipLoader loading={loading} />
      <MovieList movies={movies} />
    </Main>
  </Wrapper>
);

Home.defaultProps = {
  movies: [],
  loading: false,
};

Home.propTypes = {
  getMovies: PropTypes.func.isRequired,
  movies: PropTypes.array,
  loading: PropTypes.bool,
};

export default withMovies(Home);
