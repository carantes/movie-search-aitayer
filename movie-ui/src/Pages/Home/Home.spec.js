import React from "react";
import { shallow } from "enzyme";
import { Home } from "./Home";

describe("Home", () => {
  let home;

  beforeEach(() => {
    const dummy = () => {};
    home = shallow(<Home getMovies={dummy} />);
  });

  it("should render properly", () => {
    expect(home).toMatchSnapshot();
  });
});
