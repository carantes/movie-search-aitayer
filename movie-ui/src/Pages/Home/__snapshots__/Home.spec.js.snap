// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render properly 1`] = `
<Wrapper>
  <Header
    title="Movie Search"
  >
    <With
      onSearch={[Function]}
      startWith={3}
      waitInterval={300}
    />
  </Header>
  <Main>
    <onlyUpdateForKeys(Loader)
      color="#000000"
      css={Object {}}
      loading={false}
      size={35}
      sizeUnit="px"
    />
    <MovieList
      movies={Array []}
    />
  </Main>
</Wrapper>
`;
